src/main/java/com/cqse/teamscaleupload/TeamscaleUpload.java

```diff
@@ -233,6 +270,37 @@ public static void main(String[] args) throws Exception {
         }
     }
 
+    private static void handleSslConnectionFailure(Input input, SSLHandshakeException e) {
+        e.printStackTrace();
+        if (input.getKeyStorePath() != null) {
+            fail("Failed to connect via HTTPS to " + input.url + ": " + e.getMessage() +
+                    "\nYou enabled certificate validation and provided a keystore with certificates" +
+                    " that should be considered valid. Still, the connection failed." +
+                    " Most likely, you did not provide the correct certificates in the keystore" +
+                    " or some certificates are missing from it." +
+                    "\nPlease also ensure that your Teamscale instance is reachable under " + input.url +
+                    " and that it is configured for HTTPS, not HTTP. E.g. open that URL in your" +
+                    " browser and verify that you can connect successfully.");
+        } else if (input.validateSsl) {
+            fail("Failed to connect via HTTPS to " + input.url + ": " + e.getMessage() +
+                    "\nYou enabled certificate validation. Most likely, your certificate" +
+                    " is either self-signed or your root CA's certificate is not known to" +
+                    " teamscale-upload. Please provide the path to a keystore that contains" +
+                    " the necessary public certificates that should be trusted by" +
+                    " teamscale-upload via --trusted-keystore. You can create a Java keystore" +
+                    " with your certificates as described here:" +
+                    " https://docs.teamscale.com/howto/connecting-via-https/#using-self-signed-certificates" +
+                    "\nPlease also ensure that your Teamscale instance is reachable under " + input.url +
+                    " and that it is configured for HTTPS, not HTTP. E.g. open that URL in your" +
+                    " browser and verify that you can connect successfully.");
+        } else {
+            fail("Failed to connect via HTTPS to " + input.url + ": " + e.getMessage() +
+                    "\nPlease ensure that your Teamscale instance is reachable under " + input.url +
+                    " and that it is configured for HTTPS, not HTTP. E.g. open that URL in your" +
+                    " browser and verify that you can connect successfully.");
```

Comment:
TBH I find the deduplicated version much harder to read. I'd prefer to keep it duplicated, especially since it's only very little duplicated info and very close together (same method). Also, we can't be sure that we won't add another case that doesn't fit the pattern and then we'd have to undo the deduplication again. what do you think?